File: google/src/go/matrix/matrix.go
package function

import (
    "fmt"
    "net/http"
    "time"
    "math/rand"
    "strconv"
    "io/ioutil"
    "log"
    "encoding/json"
)

type Message struct {
    Success bool `json:"success"`
    Payload Payload `json:"payload"`
    Metrics Metrics `json:"metrics"`

}
    
type Payload struct {
    Test string `json:"test"`
    N int `json:"n"`
    Time int `json:"time"`
}

type Metrics struct {
    MachineId string `json:"machineid"`
    InstanceId string `json:"instanceid"`
    Cpu string `json:"cpu"`
    Mem string `json:"mem"`
    Uptime string `json:"uptime"`
}

func Go_matrix(w http.ResponseWriter, r *http.Request) {

    buf2, err := ioutil.ReadFile("/proc/cpuinfo")
	if err != nil {
		log.Fatal(err)
    }
    cpuinfo := string(buf2)

    buf3, err := ioutil.ReadFile("/proc/meminfo")
	if err != nil {
		log.Fatal(err)
    }
    meminfo := string(buf3)

    buf4, err := ioutil.ReadFile("/proc/uptime")
	if err != nil {
		log.Fatal(err)
    }
    uptime := string(buf4)

    n := 100

    if r.URL.Query().Get("n") != "" {
        n, err = strconv.Atoi(r.URL.Query().Get("n"))
        if err != nil {
                log.Fatal(err)
        }
    } else {
        n = 100
    }

    start := time.Now()
    matrix(n)
    elapsed := time.Since(start)

    msg := &Message{
        Success: true,
        Payload: Payload{
            Test: "matrix test",
            N: n,
            Time: int(elapsed / time.Millisecond),
        },
        Metrics: Metrics{
            MachineId: "",
            InstanceId: "",
            Cpu: cpuinfo,
            Mem: meminfo,
            Uptime: uptime,
        },
    }

    b, err := json.Marshal(msg)

    w.Header().Set("Content-Type", "application/json")
    w.WriteHeader(200)

    fmt.Fprint(w, string(b))
}

func randomTable(n int) [][]int {

    s1 := rand.NewSource(time.Now().UnixNano())
    r1 := rand.New(s1)

    m := make([][]int, n) 
    for i:=0; i<n; i++ {
        m[i] = make([]int, n)
        for j:=0; j<n; j++ {
            m[i][j] = r1.Intn(100)
        }
    }
    return m
}

func matrix(n int) [][]int {
    matrixA := randomTable(n)
    matrixB := randomTable(n)
    matrixMult := make([][]int, n)
 
    for i := 0; i < len(matrixA); i++ {
        matrixMult[i] = make([]int, n)
        for j := 0; j < len(matrixB); j++ {
            sum := 0
            for k := 0; k < len(matrixA); k++ {
                sum += matrixA[i][k] * matrixB[k][j]
            }
            matrixMult[i][j] = sum
        }
    }

    return matrixMult
}

File: azure/src/dotnet/dotnet_matrix/matrix.cs
using System;
using System.Net;
using System.Net.Http;
using System.IO;
using System.Text;
using System.Diagnostics;
using System.Threading.Tasks;
using System.Collections.Generic;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using Newtonsoft.Json.Linq;

namespace dotnet_matrix
{
    public static class matrix
    {
        [FunctionName("matrix")]
        public static async Task<HttpResponseMessage> Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
            ILogger log)
        {
            string instanceId = ""; 
            string cpuinfo = "";
            string meminfo = "";
            string uptime = "";

            if (Directory.Exists("/proc"))
            {
                instanceId = File.ReadAllText("/proc/self/cgroup");
                cpuinfo = File.ReadAllText("/proc/cpuinfo");
                meminfo = File.ReadAllText("/proc/meminfo");
                uptime = File.ReadAllText("/proc/uptime");
            }

            int n = 100;

            if(req.Query != null && req.Query.ContainsKey("n")) {
                bool parseOk = Int32.TryParse(req.Query["n"], out n);
                if(!parseOk) {
                    n = 100;
                }
            } else {
                n = 100;
            }

            Stopwatch sw = new Stopwatch();
            sw.Start();
		    int[,] result = matrixMult(n);
            sw.Stop();

            JObject message = new JObject();
            message.Add("success", new JValue(true));
            JObject payload = new JObject();
            payload.Add("test", new JValue("matrix test"));
            payload.Add("n", new JValue(n));
            payload.Add("time", new JValue(sw.Elapsed.TotalMilliseconds));
            message.Add("payload", payload);
            JObject metrics = new JObject();
            metrics.Add("machineid", new JValue(""));
            metrics.Add("instanceid", new JValue(instanceId));
            metrics.Add("cpu", new JValue(cpuinfo));
            metrics.Add("mem", new JValue(meminfo));
            metrics.Add("uptime", new JValue(uptime));
            message.Add("metrics", metrics);

            return new HttpResponseMessage(HttpStatusCode.OK) {
                Content = new StringContent(message.ToString(), Encoding.UTF8, "application/json")
            };
        }

        public static int[,] randomTable(int n) {
            Random rnd = new Random();
            int[,] m = new int[n,n];
            for(int i=0;i<n;i++)
            {
                for(int j=0;j<n;j++)
                {
                    Console.WriteLine("i: " + i);
                    Console.WriteLine("j: " + j);
                    m[i, j]= rnd.Next(1, 100);
                }
            }
            Console.WriteLine("ok");
            return m;
        }

        public static int[,] matrixMult(int n) {
            int[,] matrixA = randomTable(n);
            Console.WriteLine("1st arrray");
            int[,] matrixB = randomTable(n);
            int[,] matrixMult = new int[n, n];
            
            for (int i = 0; i < matrixA.GetLength(0); i++) {
                for (int j = 0; j < matrixB.GetLength(0); j++) {
                    int sum = 0;
                    for (int k = 0; k < matrixA.GetLength(0); k++) {
                        sum += matrixA[i,k] * matrixB[k,j];
                    }
                    matrixMult[i,j] = sum;
                }
            }

            return matrixMult;
        }

    }
}
